src/__tests__/builder.ts

```diff
@@ -1,21 +1,21 @@
 export interface IBuilder<T> {
-  with<K extends keyof T>(key: K, value: T[K]): IBuilder<T>;
+  with<K extends keyof T>(key: K, value: T[K]): this;
 
   build(): T;
 }
 
 export class Builder<T> implements IBuilder<T> {
-  private constructor(private target: Partial<T>) {}
+  constructor(private target: Partial<T>) {}
```

Comment:
We can start with `protected` as I understand this is only going to be used by subclasses.

Reply:
Adjusted in f7bfad5d54cfc2a28765809536b4989f838b2626.